### PR TITLE
`book`: Document probe-run json mode

### PR DESCRIPTION
**Pull request by Urhengulas**
Document the json output, which was implemented in #651.

Rendered version: https://deploy-preview-655--admiring-dubinsky-56dff5.netlify.app/json-output.html

### CODE REVIEW

File: book/src/json-output.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+# JSON output
+
+> Structured logging for `probe-run`.
+
+As an alternative to it's human-focused output, `probe-run` offers structured json output. There are two use-cases:
```
Comment (jonathanpallant) [inline book/src/json-output.md:5]: I think it should be JSON not json throughout. Except when it's an argument like `--json-mode`.

Comment (jonathanpallant) [inline book/src/json-output.md:5]: Its not it's. *it's* == *it is*, while *its* is the gender neutral version of *his* or *hers*.

File: book/src/json-output.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    pub modules: Vec<String>,
+    pub function: String,
+}
+
+# mod log { pub struct Level; }
+```
+
+... which results in following json schema ...
+
+```json
```
Comment (jonathanpallant) [inline book/src/json-output.md:109]: This is JSON Schema, not JSON. The standard Github JSON renderer makes a mess of it.

File: book/src/json-output.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            "modules": [],
+            "function": "__cortex_m_rt_main"
+        }
+    },
+    "target_timestamp": "1"
+}
+```
+
+## SchemaVersion
+
```
Comment (jonathanpallant) [inline book/src/json-output.md:148]: What's this for, or how would you use it?

File: book/src/json-output.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+It indicates the version of the json format you are using. `probe-run` will always output it as a header at the beginning of each stream of logs. We anticipate that the format will slightly change while `probe-run` and `defmt` evolve. Using this version you always know which revision is in use and can act upon that.
+
+🤗: Thank you for the explanation!
+
+## JsonFrame
+
+The [`defmt-json-schema`] crate provides a data transfer object `struct JsonFrame` which is used to serialize the log frame to json and can be used to deserialize it back to rust as well.
+
+The `JsonFrame` is defined like this ...
```
Comment (jonathanpallant) [inline book/src/json-output.md:81]: It might be worth an example of loading a saved JSON file and `{:?}` printing all the objects.

**Review by jonathanpallant — COMMENTED** [review]
Looks really good. Just a few minor notes.

File: book/src/json-output.md (hunk 2 above)
Comment (Urhengulas) [reply book/src/json-output.md:109]: It does look bad on github, but when rendered in the mdbook it looks good I'd say.

File: book/src/json-output.md (hunk 4 above)
Comment (Urhengulas) [reply book/src/json-output.md:81]: Like this? 😁 https://github.com/knurling-rs/defmt/blob/1dea0c1de8469ba27a3b56760a9b98f22870aaaf/decoder/defmt-json-schema/examples/simple.rs#L1-L18

I will link it here.

File: book/src/json-output.md (hunk 3 above)
Comment (Urhengulas) [reply book/src/json-output.md:148]: Are you talking about the example of the JSON output, or the SchemaVersion?

Comment (jonathanpallant) [reply book/src/json-output.md:156]: The SchemaVersion. It wasn't clear what "can be used to deserialise the JSON back to Rust" had to do with a struct containing a field called `schema_version`. So maybe an example would help?

File: book/src/json-output.md (hunk 2 above)
Comment (jonathanpallant) [reply book/src/json-output.md:117]: OK, well I guess most people are looking at the MdBook.

File: book/src/json-output.md (hunk 3 above)
Comment (jonathanpallant) [reply book/src/json-output.md:156]: Oh wait, I think I get it. Maybe add something like:

> If you wish to deserialise the entire file back into a Rust program, you will need to be able to decode the `SchemaVersion` object at the start of the file, as well as the `JsonFrame` objects. To do that, we supply a `SchemaVersion` struct in .... which you can use by ....

File: book/src/json-output.md (hunk 4 above)
Comment (jonathanpallant) [reply book/src/json-output.md]: Yes, that does help things make sense!

File: book/src/json-output.md (hunk 3 above)
Comment (Urhengulas) [reply book/src/json-output.md]: I've reformulated it. What do you think now?

Comment (jonathanpallant) [reply book/src/json-output.md]: Nice :) 👍

**Comment by Urhengulas** [comment]
bors r=@jonathanpallant

**Pull request merged**